### PR TITLE
Automatically size the Servlet buffers

### LINKED ISSUE #3881: Memory usage increase with the new network stack
When deploying the `getting-started` application in OpenShift, I have seen that the new network stack version required 24Mb or memory instead of 12Mb to serve the requests.  With 12Mb, `OutOfMemory` exception are thrown.

**How to reproduce it**

1. You need an OpenShift cluster somewhere (I used crc), and `oc` connected to it.
2. Create a new project with `oc new-project some-quarkus-test`
3. Creates a build config with:
```
oc new-build --name=my-application \
  --binary=true \
  --docker-image=quay.io/quarkus/ubi-quarkus-native-binary-s2i:19.2.0
```
4. Go to the quickstarts, start with the `master` branch, build the native image (generate a linux64 executable with `mvn clean install -Pnative -Dnative-image.docker-build=true`)
5. Creates a `target/.s2i/environment` file with:
```
QUARKUS_OPTS=-Xmx12M -Xms8M -Xmn12M
```
6. Deploy the application with:
```
oc start-build my-application --from-dir target
oc new-app -i my-application
oc expose svc/my-application
```
7. Test the application, it should work

Now, switch to the `development` branch, and rebuild the application:
`mvn install -Pnative -Dnative-image.docker-build=true` (don't clean)

8. Redeploy the application with:
```
oc start-build my-application --from-dir target
```

Test the application, it will fail.

If you edit the `QUARKUS_OPTS` application to `QUARKUS_OPTS=-Xmx24M -Xms16M -Xmn24M` and restart the build, it works.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #3881, Servlet buffers are way to big
for constrained environments. Note that memory usage
is still a bit up due to extra classes being loaded.

### CODE REVIEW

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java [hunk 1]
```diff
@@ -98,43 +100,28 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
     /**
      * TODO: configuration
      */
-    protected static final int BUFFER_SIZE = 8 * 1024;
```
Comment (gsmet) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java]: Should the TODO above go away now?

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.runtime.configuration.MemorySize;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, name = "servlet")
+public class ServletRuntimeConfig {
+
+    /**
+     * The buffer size to use for Servlet
+     */
```
Comment (gsmet) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletRuntimeConfig.java]: Maybe we should document the default behavior?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few minor comments. Nothing blocking.

**Comment by cescoffier** [comment]
I confirm that this fixes the memory issue I've seen in CRC.

**Comment by geoand** [comment]
This now also need a rebase

**Pull request merged**